src/bun.js/bindings/ModuleLoader.cpp

```diff
@@ -1136,12 +1152,12 @@ BUN_DEFINE_HOST_FUNCTION(jsFunctionOnLoadObjectResultResolve, (JSC::JSGlobalObje
     bool wasModuleMock = pendingModule->wasModuleMock;
 
     JSC::JSValue result = handleVirtualModuleResult<false>(reinterpret_cast<Zig::GlobalObject*>(globalObject), objectResult, &res, &specifier, &referrer, wasModuleMock);
+    if (scope.exception()) [[unlikely]] {
```

Comment:
```suggestion
    if (!scope.exception() && !res.success) [[unlikely]] {
      throwException(globalObject, scope, result);
    }
    if (scope.exception()) [[unlikely]] {
```

and then delete the `if (res.sucess)` statement and the `else` branch